### PR TITLE
fix(toolbar): allow sentry in CSP script-src directive

### PR DESCRIPTION
**Pull request by aliu39**
Reopened from https://github.com/getsentry/sentry/pull/78166

Adding details soon

### CODE REVIEW

File: src/sentry/toolbar/views/__init__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    csp = response.headers.get("Content-Security-Policy")
+    if csp:
+        for d in csp.split(";"):
+            d = d.strip()
+            if d.startswith("script-src "):
+                sources = d.split()[1:]
+                return all([src in sources for src in TOOLBAR_CSP_SCRIPT_SRC])
+    elif "Content-Security-Policy-Report-Only" in response.headers:
+        return True
+    return False
```
Comment (cmanallen) [inline src/sentry/toolbar/views/__init__.py:14]: This file should be renamed `test_utils` if you're only using it in the tests.

File: src/sentry/toolbar/views/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+TOOLBAR_CSP_SCRIPT_SRC = ["'unsafe-inline'", "sentry.io", "*.sentry.io"]
```
Comment (cmanallen) [inline src/sentry/toolbar/views/__init__.py:1]: This won't work for self-hosted customers.  How do we support their domain?  I'm wondering if there's a setting in `server.conf` or elsewhere that has this value.

**Review by cmanallen — CHANGES_REQUESTED** [review]
I think this looks good.  Are you happy with the amount of test coverage?

File: src/sentry/toolbar/views/__init__.py (hunk 1 above)
Comment (aliu39) [reply src/sentry/toolbar/views/__init__.py:17]: The TOOLBAR_SCRIPT_SRC is used by src and tests. The function is only for tests though.

**Comment by aliu39** [comment]
Ref https://github.com/getsentry/sentry/issues/78237

**Pull request closed without merging**